pkg/auth/service.go

```diff
@@ -778,152 +753,148 @@ func (s *DBAuthService) AddCredentials(ctx context.Context, username, accessKeyI
+	err := s.store.DeleteMsg(ctx, model.PartitionKey, pg)
+	if err != nil {
+		return fmt.Errorf("policy detachment to group: (key %s): %w", pg, err)
+	}
 	return err
 }
 
-func (s *DBAuthService) GetCredentialsForUser(ctx context.Context, username, accessKeyID string) (*model.Credential, error) {
-	credentials, err := s.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		if _, err := getUser(tx, username); err != nil {
-			return nil, err
-		}
-		credentials := &model.DBCredential{}
-		err := tx.Get(credentials, `
-			SELECT auth_credentials.*
-			FROM auth_credentials
-			INNER JOIN auth_users ON (auth_credentials.user_id = auth_users.id)
-			WHERE auth_credentials.access_key_id = $1
-				AND auth_users.display_name = $2`, accessKeyID, username)
-		if err != nil {
-			return nil, err
-		}
-		return credentials, nil
-	})
+func (s *KVAuthService) GetCredentialsForUser(ctx context.Context, username, accessKeyID string) (*model.Credential, error) {
+	if _, err := s.GetUser(ctx, username); err != nil {
+		return nil, fmt.Errorf("%s: %w", username, err)
+	}
+	creadantialsKey := model.KvCredentialPath(username, accessKeyID)
+	m := model.CredentialData{}
+	_, err := s.store.GetMsg(ctx, model.PartitionKey, creadantialsKey, &m)
 	if err != nil {
 		return nil, err
 	}
-	return model.ConvertCreds(credentials.(*model.DBCredential)), nil
+	m.SecretAccessKey = ""
+	return model.CredentialFromProto(&m), nil
 }
 
-func (s *DBAuthService) GetCredentials(ctx context.Context, accessKeyID string) (*model.Credential, error) {
-	return s.cache.GetCredential(accessKeyID, func() (*model.Credential, error) {
-		credentials, err := s.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-			credentials := &model.DBCredential{}
-			err := tx.Get(credentials, `SELECT * FROM auth_credentials WHERE auth_credentials.access_key_id = $1`,
-				accessKeyID)
-			if err != nil {
-				return nil, err
-			}
-			key, err := s.decryptSecret(credentials.SecretAccessKeyEncryptedBytes)
-			if err != nil {
-				return nil, err
-			}
-			credentials.SecretAccessKey = key
-			return credentials, nil
-		})
-		if err != nil {
+func (s *KVAuthService) GetCredentials(ctx context.Context, accessKeyID string) (*model.Credential, error) {
+	m := &model.UserData{}
+	itr, err := s.store.Scan(ctx, m.ProtoReflect().Type(), model.PartitionKey, model.KvUserPath(""))
+	if err != nil {
+		return nil, fmt.Errorf("sacn users users: %w", err)
+	}
+	defer itr.Close()
+
+	for itr.Next() {
+		if itr.Err() != nil {
+			return nil, itr.Err()
+		}
+		entry := itr.Entry()
+		user, ok := entry.Value.(*model.UserData)
+		if !ok {
+			return nil, fmt.Errorf("list users: %w", err)
+		}
+		c := model.CredentialData{}
+		credentialsKey := model.KvCredentialPath(user.Username, accessKeyID)
+		_, err := s.store.GetMsg(ctx, model.PartitionKey, credentialsKey, &c)
+		if err != nil && errors.Is(err, kv.ErrNotFound) {
 			return nil, err
+		} else if !errors.Is(err, kv.ErrNotFound) {
```

Comment:
why do you check it again?

Reply:
Because If the credentials doesn't exist I don't want to exit I want to continue until I found the user that has these credentials